src/common/leaky_bucket.cpp

```diff
@@ -79,29 +86,58 @@ double LeakyBucket::Bucket::Add(double tokens) {
     }
 }
 
-void LeakyBucket::Bucket::Leak(uint64_t intervalUs) {
+void LeakyBucket::Bucket::Leak() {
     if (avg == 0) {
         level = 0;
         burstLevel = 0;
         return;
     }
 
-    double leak = static_cast<double>(avg) * intervalUs /
-                  TimeUtility::MicroSecondsPerSecond;
-    level = std::max(level - leak, 0.0);
+    if (level == 0) {
```

Comment:
level is double, you can't directly compare it with zero

Reply:
fix